dmutils/email/tokens.py

```diff
@@ -153,7 +149,7 @@ def decode_invitation_token(encoded_token):
 
             return {
                 'error': 'token_expired',
-                'role': token.get('role') or ('supplier' if token.get('supplier_id') else 'buyer')
```

Comment:
Thinking about it, I think this line may need to stay as is. It's possible that someone still has an email with a link/token in it that was generated before we switched to the new token styles. 

If they click that link, the old token will throw an InvalidToken error when decrypted. We'll then try and decrypt it again with no max age, which will work, but we'll set the role key here to `None`. Which will then get passed to the user-frontend and the template will get rendered without user role specific information, which would be bad.

Reply:
Good point, I'll change it back.

Reply:
Do you reckon we need a new test for that scenario?

Reply:
I think we already have them, they're the second two tests you're removing in this PR, so should probably keep them.